include/revng/RestructureCFG/ScopeGraphGraphTraits.h

```diff
@@ -306,6 +307,90 @@ struct llvm::GraphTraits<llvm::Inverse<Scope<const llvm::Function *>>>
   }
 };
 
+inline std::string getNodeLabel(const llvm::BasicBlock *N,
+                                const llvm::Function *F) {
+  if (N->hasName()) {
+    return N->getName().str();
+  } else {
+
+    // We iterate over the function and we assign an incremental ID for each
+    // unnamed `BasicBlock` that we encounter
+    unsigned ID = 0;
+    for (const auto *Node : llvm::nodes(F)) {
+      if (N == Node) {
+        return "block_" + std::to_string(ID);
+      } else {
+        ++ID;
+      }
+    }
+  }
+
+  revng_abort("We should always find the basicblock in the loop above");
+}
+
+inline std::string
+getEdgeAttributes(const llvm::BasicBlock *N,
+                  llvm::GraphTraits<
+                    Scope<llvm::BasicBlock *>>::ChildIteratorType EI,
+                  const Scope<llvm::Function *> G) {
+
+  // If we are printing a block which has a `scope_closer` edge, which by
+  // construction must be unique, if present, and the last in the
+  // `MaterializedRange` representing the successors, we print such edge as
+  // dashed
+  if (getScopeCloserTarget(N) and EI.atLastElement()) {
```

Comment:
I'm not excited by `atLastElement`. Can't you check if the target block of EI is the scope closer target of `N` instead?

If so, I'd also drop the `atLastElement` method.

Reply:
I fixed this with the solution we discussed IRL, not the one proposed in this comment:
https://github.com/revng/revng/blob/10dcf9302e9e2d88a344b05b9764d19909fd31a9/include/revng/RestructureCFG/ScopeGraphGraphTraits.h#L342

Dropped the `atLastElement` method from the `EagerMaterializationRangeIterator`